packages/indexer-agent/src/network.ts

```diff
@@ -1365,11 +1369,23 @@ export class Network {
         },
       )
 
-      const allocationIds = allocations.map(allocation => allocation.id)
+      // Max claims per batch should roughly be equal to average gas per claim / block gas limit
+      // On-chain data shows an average of 120k gas per claim and the block gas limit is 15M
+      // If we add a 30k gas buffer (for 150k estimated gas per claim)
+      // Then we could fit a maximum of a 100 claim batch in a block
+      const MAX_CLAIMS_PER_BATCH = 100
+      const allocationIds = allocations
+        .map(allocation => allocation.id)
+        .slice(0, MAX_CLAIMS_PER_BATCH)
```

Comment:
So talking through this: 
In the case that there are more than 100 claims in a potential batch this will limit it to 100 and discard the rest.  Those "discarded" allocations will then be processed on the next loop?

Reply:
That's right. It's also the reason I added the second log to clarify the allocation IDs that were included in the batch.